qiita_ware/processing_pipeline.py

```diff
@@ -254,6 +254,40 @@ def _get_preprocess_fasta_cmd(raw_data, prep_template, params):
                         "-o %s" % output_dir,
                         params_str])
     else:
+        len_seqs = len(seqs)
+        len_mapping_fps = len(mapping_fps)
+
+        if len_mapping_fps > len_seqs:
+            mapping_fps = [basename(m) for m in mapping_fps]
+            sffs = [basename(s) for s in sffs]
+            raise ValueError(
+                'Your prep template defines: "%s" but you only have "%s" as '
+                'files' % (', '.join(mapping_fps), ', '.join(sffs)))
+
+        if len_seqs != len_mapping_fps:
+            # -8 is to remove the _MMF.txt
+            prefixes = {m: {'prefix': basename(m)[:-8], 'seqs': [],
+                            'quals': []} for m in mapping_fps}
+            counter = 0
+            for p in prefixes:
```

Comment:
relooking at this, I think this can go from a nested forloop to a single forloop. Right now, the loops are: for every prefix, and for every sequence file, do something. However, the loop structure could instead be: for every sequence file, determine prefix and do something. It'll reduce the nesting and is less computational effort